docs/resources/vmdistributedcluster.md

```diff
@@ -0,0 +1,120 @@
+
+A `VMClusterRefOrSpec` object has the following mutually exclusive fields:
+
+*   `ref`: A `corev1.LocalObjectReference` pointing to an existing `VMCluster` object by name. If `ref` is specified, `name` and `spec` are ignored.
+*   `overrideSpec`: An optional JSON object that specifies an override to the `VMClusterSpec` of the referenced object when `ref` is used. This allows applying common configuration changes across referenced clusters without modifying their original definitions.
+*   `name`: A static name to be used for the `VMCluster` when `spec` is provided. This field is ignored if `ref` is specified.
+*   `spec`: A `vmv1beta1.VMClusterSpec` object that defines the desired state of a new `VMCluster`. This field is ignored if `ref` is specified.
+
+**Example: Defining a `VMDistributedCluster` with inline `VMCluster` specifications for two zones:**
+
+```yaml
+apiVersion: operator.victoriametrics.com/v1alpha1
+kind: VMDistributedCluster
+metadata:
+  name: my-distributed-cluster
+spec:
+  vmagent:
+    name: my-distributed-vmagent
+  vmusers:
+    - name: my-read-user-for-zone-a
+    - name: my-read-user-for-zone-b
+  zones:
+    vmclusters:
+      - name: zone-a
+        spec:
+          vmstorage:
+            replicaCount: 1
+            storage:
+              volumeClaimTemplate:
+                spec:
+                  resources:
+                    requests:
+                      storage: 10Gi
+          vmselect:
+            replicaCount: 1
+          vminsert:
+            replicaCount: 1
+      - name: zone-b
+        spec:
+          vmstorage:
+            replicaCount: 1
+            storage:
+              volumeClaimTemplate:
+                spec:
+                  resources:
+                    requests:
+                      storage: 10Gi
+          vmselect:
+            replicaCount: 1
+          vminsert:
+            replicaCount: 1
+  paused: false
+```
+
+**Example: Referencing existing `VMCluster` objects and applying an `overrideSpec`:**
+
+```yaml
+apiVersion: operator.victoriametrics.com/v1alpha1
+kind: VMDistributedCluster
+metadata:
+  name: referenced-distributed-cluster
+spec:
+  zones:
+    - vmclusters:
+        ref:
+          name: existing-cluster-zone-1
+        overrideSpec:
+          vmstorage: # Increase PVC for vmstorage in this zone
+            storage:
+              volumeClaimTemplate:
+                spec:
+                  resources:
+                    requests:
+                      storage: 20Gi
+    - vmclusters:
+        ref:
+          name: existing-cluster-zone-2
+        overrideSpec:
+          vmstorage: # Increase vmstorage replicas for this zone
+            replicaCount: 2
+```
+
+### Current shortcomings
+- When VMDistributedCluster is removed, all resources created inline are not deleted.
```

Comment:
```suggestion
- If a VMCluster is created by VMDistributedCluster, its resources won’t be deleted when VMDistributedCluster is deleted.
```
Since the inline resources only include vmcluster, let's make it clearer.